### ISSUE TITLE
TextArea component is missing from react-aria-components

### ISSUE THREAD

**Issue by snowystinger** [adobe/react-spectrum#4595]
### Discussed in https://github.com/adobe/react-spectrum/discussions/4594

<div type='discussions-op-text'>

<sup>Originally posted by **eashish93** May 25, 2023</sup>
Playing with react-aria-components for a mvp project and found that `TextArea` component is missing from the doc. In the meantime I'm using my own created with hooks version. It's strange it's not in alpha because it's literally a copy of `Input` element. 
</div>

**Comment by rvramesh** [comment]
I can help here. Any guidance on the implementation?

**Comment by devongovett** [comment]
One question is if we should have a whole separate `<TextArea>` component or just add an `elementType` prop to `<Input>` since they are usually used in the same places?

**Comment by rvramesh** [comment]
I am of the opinion that both should be seperate. From the https://github.com/DefinitelyTyped/DefinitelyTyped/blob/3197efc097d522c4bf02b94e1a0766d007d6cdeb/types/react/v17/index.d.ts#L2189 and https://github.com/DefinitelyTyped/DefinitelyTyped/blob/3197efc097d522c4bf02b94e1a0766d007d6cdeb/types/react/v17/index.d.ts#L2398 the props which are applicable for input may not be applicable for text area and vice versa. So, would suggest to create it as a separate component.

**Comment by rvramesh** [comment]
May be internally call the same component with element type and props but expose two components as done in react spectrum?

**Comment by devongovett** [comment]
hmm yeah the types are a good point. Which components were you expecting to be able to use a TextArea within rather than an Input? Only within a TextField, or should it also be supported in other places like SearchField, ComboBox, NumberField, etc.? I could see some of those being a little strange, e.g. conflicting Enter key behavior.

Also the top-level component will need to know what type of element it contains in order to send correct props. For example, the `useTextField` hook currently accepts an `inputElementType` prop which can be set to `input` or `textarea`. This changes what props it includes, such as `type` and `pattern` which aren't supported on textarea.
